docs/resources/deployment.md

```diff
@@ -0,0 +1,67 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "juju_deployment Resource - terraform-provider-juju"
+subcategory: ""
+description: |-
+  A resource that represents a Juju deployment.
+---
+
+# juju_deployment (Resource)
+
+A resource that represents a Juju deployment.
+
+## Example Usage
+
+```terraform
+resource "juju_deployment" "this" {
+  name = "foobar" # optional, set to charm name when absent
+
+  model = juju_model.development.uuid # required, model uuid
```

Comment:
Working with the model UUID is not going to be the best UX although I think it will be OK for a first iteration.

Reply:
Can we just pass `juju_model.development` here and have the provider automatically pickup the UUID?